llvm/lib/Target/RISCV/GISel/RISCVCallLowering.cpp

```diff
@@ -303,6 +309,11 @@ static bool isSupportedArgumentType(Type *T, const RISCVSubtarget &Subtarget) {
     return true;
   if (T->isPointerTy())
     return true;
+  // TODO: support 16bit FPtypes.
+  // TODO: Support fixed vector types.
+  if (IsLowerArgs && T->isVectorTy() && Subtarget.hasVInstructions() &&
+      !T->is16bitFPTy() && T->isScalableTy())
```

Comment:
I'm not sure why've singled out 16BitFPTy here. Aren't vectors of i128 or f128 also not supported or any other random integer type that's isn't i8/i16/i32/i64? Also vectors of i64 aren't supported with only the Zve32* extensions.

Reply:
You make a good point. I think it may make sense to support for these kinds of types (including 16bitfp) in this patch.